orc8r/cloud/go/services/orchestrator/servicers/builder_servicer_test.go

```diff
@@ -32,6 +33,93 @@ import (
 	mconfig_protos "magma/orc8r/lib/go/protos/mconfig"
 )
 
+func testUtilForTestSharedConfig(t *testing.T, orc8rVersion string) {
```

Comment:
Nit: files should generally read top-down, meaning, generally, when I "jump to reference" I'm moving further down in a file. This makes fore a more uniform reading experience

Here specifically, it's offputting to see a util at the top of a test file, since casual readers care principally about the actual tests. So let's drop this guy at least after the tests he's used in

Reply:
it was done to avoid code duplication, which almost never is a good idea. this util function makes for a smaller code footprint, avoid any problems that occur with code duplication.

Reply:
To clarify: I'm definitely supportive of having this fn exist 🙂  Just preference to move its declaration down beneath the fns that call it

Reply:
Correct that in C/C++ context this is expected. However in Go-land, this is not a specific requirement, so we are free to make a decision based on readability. In my experience, and with the goal of the code being approachable, a top-down approach is preferable -- i.e. provide summary first, then fill in some details, then finally provide lowest-layer details

For example, when debugging Go code, I read the golang code first, and only drop down into the generated assembly if a higher-resolution understanding is required. Similarly, when reading Go code, I want to be able to skim a summary of a file by looking just at the top part of the file, delving down further in the file if a higher-resolution understanding is required. Finally, consider e.g. http://gaia.cs.umass.edu/kurose_ross/online_lectures.htm, which breaks an entire field's intro class into progressively more-detailed descriptions, starting with the highest level and then progressively filling in the missing details

That to say -- when given the choice of whether to put (a) the finest details or (b) the summary first, I'd argue it's most natural to go with option (b). Which would then point to the idea that e.g. this private function definition should be placed after its usage, because this private function definition provides impl details rather than a summary of this file's purpose

Reply:
In any case, given this is a small consideration, I think default is to go with the existing style (as much as it exists), which is option (b). If you feel strongly enough to change the overarching style, we can handle it in a future PR 🙂